apps/studio/src/components/connection/CommonEntraId.vue

```diff
@@ -0,0 +1,262 @@
+          type="text"
+          class="form-control"
+          v-model="username"
+        />
+      </div>
+      <div class="form-group" v-show="showPassword">
+        <label for="password">Password</label>
+        <input
+          name="password"
+          type="text"
+          class="form-control"
+          v-model="config.password"
+        />
+      </div>
+      <div class="form-group" v-show="showTenantId">
+        <label for="tenantId">
+          Tenant ID
+          <i
+            class="material-icons"
+            style="padding-left: 0.25rem"
+            v-tooltip="{
+              content:
+                'This can be found in the <code>\'Microsoft Entra ID\'</code> section of Azure, <br/> in the Overview labelled <code>\'Tenant ID\'</code>',
+              html: true,
+            }"
+            >help_outlined</i
+          >
+        </label>
+        <input
+          name="tenantId"
+          type="text"
+          class="form-control"
+          v-model="config.azureAuthOptions.tenantId"
+        />
+      </div>
+      <div class="form-group" v-show="showClientSecret">
+        <label for="clientSecret">Client Secret</label>
+        <input
+          name="clientSecret"
+          type="text"
+          class="form-control"
+          v-model="config.azureAuthOptions.clientSecret"
+        />
+      </div>
+      <div class="form-group" v-show="showMsiEndpoint">
+        <label for="msiEndpoint">MSI Endpoint</label>
+        <input
+          name="msiEndpoint"
+          type="text"
+          class="form-control"
+          v-model="config.azureAuthOptions.msiEndpoint"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+<script>
+import { AzureAuthType } from "@/lib/db/types";
+import { AppEvent } from "@/common/AppEvent";
+import _ from "lodash";
+import { onMounted } from "vue";
+
+export default {
+  props: ["config", "authType"],
+  data() {
+    return {
+      azureAuthEnabled: this.config.azureAuthOptions?.azureAuthEnabled,
+      accountName: null,
+      signingOut: false,
+      errorSigningOut: null,
+    };
+  },
+  computed: {
+    username: {
+      get() {
+        return this.config.username || this.config.user;
+      },
+      set(value) {
+        this.config.username = value;
+        this.config.user = value;
+      },
+    },
+    showUser() {
+      return [AzureAuthType.Password, AzureAuthType.CLI].includes(
+        this.authType
+      );
+    },
+    showPassword() {
+      return [AzureAuthType.Password].includes(this.authType);
+    },
+    showTenantId() {
+      return [
+        AzureAuthType.Password,
+        AzureAuthType.ServicePrincipalSecret,
+      ].includes(this.authType);
+    },
+    showClientSecret() {
+      return [AzureAuthType.ServicePrincipalSecret].includes(this.authType);
+    },
+    showMsiEndpoint() {
+      return [AzureAuthType.MSIVM].includes(this.authType);
+    },
+    showCli() {
+      return [AzureAuthType.CLI].includes(this.authType);
```

Comment:
If it just has one element in the array, I'd rather we just check if they are equal.
```suggestion
      return AzureAuthType.ServicePrincipalSecret === this.authType;
```